esphome/components/seeed_mr24hpc1/number/__init__.py

```diff
@@ -0,0 +1,132 @@
+from esphome.const import (
+    ENTITY_CATEGORY_CONFIG,
+)
+from .. import CONF_MR24HPC1_ID, MR24HPC1Component, mr24hpc1_ns
+
+SensitivityNumber = mr24hpc1_ns.class_("SensitivityNumber", number.Number)
+CustomModeNumber = mr24hpc1_ns.class_("CustomModeNumber", number.Number)
+ExistenceThresholdNumber = mr24hpc1_ns.class_("ExistenceThresholdNumber", number.Number)
+MotionThresholdNumber = mr24hpc1_ns.class_("MotionThresholdNumber", number.Number)
+MotionTriggerTimeNumber = mr24hpc1_ns.class_("MotionTriggerTimeNumber", number.Number)
+Motion2RestTimeNumber = mr24hpc1_ns.class_("Motion2RestTimeNumber", number.Number)
+CustomUnmanTimeNumber = mr24hpc1_ns.class_("CustomUnmanTimeNumber", number.Number)
+
+CONF_SENSITIVITY = "sensitivity"
+CONF_CUSTOM_MODE = "custom_mode"
+CONF_EXISTENCE_THRESHOLD = "existence_threshold"
+CONF_MOTION_THRESHOLD = "motion_threshold"
+CONF_MOTION_TRIGGER = "motion_trigger"
+CONF_MOTION_TO_REST = "motion_to_rest"
+CONF_CUSTOM_UNMAN_TIME = "custom_unman_time"
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(CONF_MR24HPC1_ID): cv.use_id(MR24HPC1Component),
+        cv.Optional(CONF_SENSITIVITY): number.number_schema(
+            SensitivityNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:archive-check-outline",
+        ),
+        cv.Optional(CONF_CUSTOM_MODE): number.number_schema(
+            CustomModeNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:cog",
+        ),
+        cv.Optional(CONF_EXISTENCE_THRESHOLD): number.number_schema(
+            ExistenceThresholdNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+        ),
+        cv.Optional(CONF_MOTION_THRESHOLD): number.number_schema(
+            MotionThresholdNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+        ),
+        cv.Optional(CONF_MOTION_TRIGGER): number.number_schema(
+            MotionTriggerTimeNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:camera-timer",
+            unit_of_measurement="ms",
+        ),
+        cv.Optional(CONF_MOTION_TO_REST): number.number_schema(
+            Motion2RestTimeNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:camera-timer",
+            unit_of_measurement="ms",
+        ),
+        cv.Optional(CONF_CUSTOM_UNMAN_TIME): number.number_schema(
+            CustomUnmanTimeNumber,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon="mdi:camera-timer",
+            unit_of_measurement="s",
+        ),
+    }
+)
+
+
+async def to_code(config):
+    mr24hpc1_component = await cg.get_variable(config[CONF_MR24HPC1_ID])
+    if sensitivity_config := config.get(CONF_SENSITIVITY):
+        n = await number.new_number(
+            sensitivity_config,
+            min_value=0,
+            max_value=3,
+            step=1,
+        )
+        await cg.register_parented(n, config[CONF_MR24HPC1_ID])
```

Comment:
Since you already have the parent you can use it here as the second argument to save it doing a lookup again
```suggestion
        await cg.register_parented(n, mr24hpc1_component)
```